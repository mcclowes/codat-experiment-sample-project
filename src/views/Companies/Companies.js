import React from "react";
import Page from "../../components/Page";
import { Text, Loader } from "../../components/Primitives";

const Companies = (props) => {
  return (
    <Page>
      <div>
        <Text.Header>Companies</Text.Header>

        <Text>
          <a
            href="https://link.codat.io/client/042399f5-d104-4f38-9ce8-cac3524f4e88?user=0cdff739-aeb9-40dd-87ee-0e0bc17184c8"
            target="_blank"
          >
            Invite URL
          </a>
        </Text>
      </div>

      <div>
        {props.companies ? (
          props.companies.map((company, i) => {
            return (
              <Text key={i}>
                {company.name} -{" "}
                <a
                  href={`https://app.codat.io/assess/${company.id}`}
                  target="_blank"
                >
                  See details
                </a>
              </Text>
            );
          })
        ) : (
          <Loader />
        )}
      </div>
    </Page>
  );
};

export default Companies;
