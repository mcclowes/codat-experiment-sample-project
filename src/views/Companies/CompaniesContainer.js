import React, { useState, useEffect } from "react";
import Companies from "./Companies";

//const baseUrl = "http://0.0.0.0:8080/https://api.codat.io/companies?page=1&pageSize=50";
const baseUrl = "https://api.codat.io/companies?page=1&pageSize=50";

const fetchCompanies = async (callback) => {
  const encodedKey = Buffer.from(process.env.REACT_APP_API_KEY).toString(
    "base64"
  );

  const result = await fetch(baseUrl, {
    method: "GET",
    headers: {
      authorization: "Basic " + encodedKey,
      "Access-Control-Allow-Origin": "*",
      "Access-Control-Allow-Headers":
        "Origin, X-Requested-With, Content-Type, Accept",
    },
  }).then((response) => response.json());

  callback(result.results);
};

const CompaniesContainer = (props) => {
  const [companies, setCompanies] = useState(undefined);

  useEffect(() => {
    fetchCompanies(setCompanies);
  }, []);

  return <Companies companies={companies} />;
};

export default CompaniesContainer;
