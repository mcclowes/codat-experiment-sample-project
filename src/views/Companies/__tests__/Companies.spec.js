import Companies from "../Companies";
import React from "react";
import { shallow } from "enzyme";

const defaultProps = {};

const renderComponent = (newProps) => {
  const props = {
    ...defaultProps,
    ...newProps,
  };

  return shallow(<Companies {...props} />);
};

let wrapper;

it("renders component", () => {
  wrapper = renderComponent();

  expect(wrapper).toMatchSnapshot();
});
