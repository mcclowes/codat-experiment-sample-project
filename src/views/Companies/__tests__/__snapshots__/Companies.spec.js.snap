// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component 1`] = `
<Page>
  <div>
    <csx__Header
      theme={
        Object {
          "colors": Object {
            "black": "#333333",
          },
        }
      }
    >
      Companies
    </csx__Header>
    <Text>
      <a
        href="https://link.codat.io/client/042399f5-d104-4f38-9ce8-cac3524f4e88?user=0cdff739-aeb9-40dd-87ee-0e0bc17184c8"
        target="_blank"
      >
        Invite URL
      </a>
    </Text>
  </div>
  <div>
    <Loader />
  </div>
</Page>
`;
